sktime/regression/tests/test_base.py

```diff
@@ -0,0 +1,421 @@
+    """Dummy Deep Regressor to test serialization capabilities."""
+
+    def __init__(
+        self,
+        optimizer,
+    ):
+        super().__init__()
+        self.optimizer = optimizer
+
+    def build_model(self, input_shape, **kwargs):
+        return None
+
+    def _fit(self, X, y):
+        return self
+
+
+class _DummyHandlesAllInput(BaseRegressor):
+    """Dummy regressor for testing base class fit/predict."""
+
+    _tags = {
+        "capability:multivariate": True,
+        "capability:unequal_length": True,
+        "capability:missing_values": True,
+    }
+
+    def _fit(self, X, y):
+        """Fit dummy."""
+        return self
+
+    def _predict(self, X):
+        """Predict dummy."""
+        return self
+
+
+class _DummyConvertPandas(BaseRegressor):
+    """Dummy regressor for testing base class fit/predict."""
+
+    _tags = {
+        "X_inner_mtype": "nested_univ",  # which type do _fit/_predict, support for X?
+    }
+
+    def _fit(self, X, y):
+        """Fit dummy."""
+        return self
+
+    def _predict(self, X):
+        """Predict dummy."""
+        return self
+
+
+multivariate_message = r"multivariate series"
+missing_message = r"missing values"
+unequal_message = r"unequal length series"
+
+
+def test_base_regressor_fit():
+    """Test function for the BaseRegressor class fit.
+
+    Test fit. It should:
+    1. Work with 2D, 3D and DataFrame for X and nparray for y.
+    2. Record the fit time.
+    3. Have self.n_jobs set or throw  an exception if the regressor can
+    multithread.
+    4. Set the class dictionary correctly.
+    5. Set is_fitted after a call to _fit.
+    6. Return self.
+    """
+    dummy = _DummyRegressor()
+    cases = 5
+    length = 10
+    test_X1 = np.random.uniform(-1, 1, size=(cases, length))
+    test_X2 = np.random.uniform(-1, 1, size=(cases, 2, length))
+    test_X3 = _create_example_dataframe(cases=cases, dimensions=1, length=length)
+    test_X4 = _create_example_dataframe(cases=cases, dimensions=3, length=length)
+    test_y1 = np.random.randint(0, 2, size=(cases))
+    test_y2 = pd.DataFrame({"0": [1] * cases, "1": [0] * cases})
+    result = dummy.fit(test_X1, test_y1)
+    assert result is dummy
+    result = dummy.fit(test_X3, test_y2)
+    assert result is dummy
+    with pytest.raises(ValueError, match=multivariate_message):
+        result = dummy.fit(test_X2, test_y1)
+    assert result is dummy
+    result = dummy.fit(test_X3, test_y1)
+    assert result is dummy
+    with pytest.raises(ValueError, match=multivariate_message):
+        result = dummy.fit(test_X4, test_y1)
+    assert result is dummy
+
+
+TF = [True, False]
+
+
+@pytest.mark.parametrize("missing", TF)
+@pytest.mark.parametrize("multivariate", TF)
+@pytest.mark.parametrize("unequal", TF)
+def test_check_capabilities(missing, multivariate, unequal):
+    """Test the checking of capabilities.
+
+    There are eight different combinations to be tested with a regressor that can
```

Comment:
Question: if the tests are independent,  at least the failure cases, will it make sense to have them separately for each capabilities?

Reply:
Do you mean something like:

```
def test_check_missing_capabilities(missing):
    with pytest.raises(ValueError, match=missing_message):
            handles_none._check_capabilities(X_metadata)
    with pytest.warns(UserWarning, match=missing_message):
            handles_none_composite._check_capabilities(X_metadata)
```
I am not opposed to seperating these tests, may make debugging a little more straightforward. Thoughts?

Reply:
I'd consider fixing this out of scope for this PR, since this is currently copy-paste from classification.

If we fix it here, we should fix in both places, and in any case it's pre-existing code.